src/internal/m365/collection/drive/collections_tree_test.go

```diff
@@ -912,64 +956,318 @@ func (suite *CollectionsTreeUnitSuite) TestCollections_PopulateTree() {
 	}
 	for _, test := range table {
 		suite.Run(test.name, func() {
-			t := suite.T()
-
-			ctx, flush := tester.NewContext(t)
-			defer flush()
+			runPopulateTreeTest(suite.T(), drv, test)
+		})
+	}
+}
 
-			mockDrivePager := &apiMock.Pager[models.Driveable]{
-				ToReturn: []apiMock.PagerResult[models.Driveable]{
-					{Values: []models.Driveable{drv}},
-				},
-			}
+// TODO: remove when unifying test tree structs
+type populateTreeTestMulti struct {
+	name       string
+	enumerator mock.EnumerateDriveItemsDelta
+	tree       *folderyMcFolderFace
+	limiter    *pagerLimiter
+	expect     populateTreeExpected
+}
 
-			mbh := mock.DefaultDriveBHWith(user, mockDrivePager, test.enumerator)
-			c := collWithMBH(mbh)
-			counter := count.New()
+// this test focuses on quirks that can only arise from cases that occur across
+// multiple delta enumerations.
+// It is not concerned with unifying previous paths or post-processing collections.
+func (suite *CollectionsTreeUnitSuite) TestCollections_PopulateTree_multiDelta() {
+	drv := models.NewDrive()
+	drv.SetId(ptr.To(id(drive)))
+	drv.SetName(ptr.To(name(drive)))
 
-			_, err := c.populateTree(
-				ctx,
-				test.tree,
-				drv,
-				id(delta),
-				test.limiter,
-				counter,
-				fault.New(true))
+	table := []populateTreeTestMulti{
+		{
+			name: "sanity case: normal enumeration split across multiple deltas",
+			tree: newFolderyMcFolderFace(nil, rootID),
+			enumerator: mock.DriveEnumerator(
+				mock.Drive(id(drive)).With(
+					mock.Delta(id(delta), nil).With(pagesOf(pageItems(
+						driveItem(id(folder), name(folder), parentDir(), rootID, isFolder),
+						driveItem(id(file), name(file), parentDir(name(folder)), id(folder), isFile)))...),
+					mock.Delta(id(delta), nil).With(pagesOf(pageItems(
+						driveItem(idx(folder, "sib"), namex(folder, "sib"), parentDir(), rootID, isFolder),
+						driveItem(idx(file, "fsib"), namex(file, "fsib"), parentDir(namex(folder, "sib")), idx(folder, "sib"), isFolder)))...),
```

Comment:
nit: I think maybe a copy/paste error for the const at the end?
```suggestion
						driveItem(idx(file, "fsib"), namex(file, "fsib"), parentDir(namex(folder, "sib")), idx(folder, "sib"), isFile)))...),
```

Reply:
Yup!  Good catch.  It'll go away next PR either way, so I'll let that resolve it.